### PR TITLE
set container accessible selinux context on CERT_DIR

### LINKED ISSUE #38165: The cfssl container run in hack/local-up-cluster.sh gets permissions errors
Get various config errors after rebasing to latest k8s and running `local-up-cluster.sh`:
```
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-kubelet-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-kube-proxy-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-controller-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-scheduler-key.pem': No such file or directory
```
with the end result being that `kubectl get nodes` shows no resources.

Version:
```
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0+776c994",
GitCommit:"776c994", GitTreeState:"clean", BuildDate:"2016-11-03T21:47:56Z",
GoVersion:"go1.6.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6+",
GitVersion:"v1.6.0-alpha.0.1439+608df651db3545-dirty",
GitCommit:"608df651db3545b1171f424fa52e245a1449c807", GitTreeState:"dirty",
BuildDate:"2016-12-05T21:20:10Z", GoVersion:"go1.6.1", Compiler:"gc",
Platform:"linux/amd64"}
```

`hack/local-up-cluster.sh` function create_client_certkey() runs the cfssl container bind-mounted to /var/run/kubernetes.  Shelling into this container you'll see there are no `owner:group` for the files in its /cert dirt (which is the host's /var/run/kubernetes dir):
```
root@a5bfa7d640f9:/certs# ls -l    #pwd is /cert
ls: cannot access client-ca.crt: Permission denied
ls: cannot access client-ca.key: Permission denied
ls: cannot access auth-proxy-client-ca.crt: Permission denied
ls: cannot access auth-proxy-client-ca.key: Permission denied
ls: cannot access scheduler.kubeconfig: Permission denied
ls: cannot access controller.kubeconfig: Permission denied
ls: cannot access kube-proxy.kubeconfig: Permission denied
ls: cannot access kubelet.kubeconfig: Permission denied
ls: cannot access apiserver.key: Permission denied
ls: cannot access apiserver.crt: Permission denied
ls: cannot access client-ca-bundle.crt: Permission denied
ls: cannot access auth-proxy-client-ca-config.json: Permission denied
ls: cannot access client-ca-config.json: Permission denied
total 0
-????????? ? ? ? ?            ? apiserver.crt
-????????? ? ? ? ?            ? apiserver.key
-????????? ? ? ? ?            ? auth-proxy-client-ca-config.json
-????????? ? ? ? ?            ? auth-proxy-client-ca.crt
-????????? ? ? ? ?            ? auth-proxy-client-ca.key
-????????? ? ? ? ?            ? client-ca-bundle.crt
-????????? ? ? ? ?            ? client-ca-config.json
-????????? ? ? ? ?            ? client-ca.crt
-????????? ? ? ? ?            ? client-ca.key
-????????? ? ? ? ?            ? controller.kubeconfig
-????????? ? ? ? ?            ? kube-proxy.kubeconfig
-????????? ? ? ? ?            ? kubelet.kubeconfig
-????????? ? ? ? ?            ? scheduler.kubeconfig
root@a5bfa7d640f9:/certs# cat client-ca-config.json
cat: client-ca-config.json: Permission denied
```
If I run the cfssl container with `--privileged` it all works both when shelling in (root is owner and group), and `kubectl get nodes` works as expected.

Q: is there something wrong with my local env on the latest kube? Or, should the cfssl container be run privileged now? Or is there a smaller hammer?

### PR DESCRIPTION
**Pull request by sjenning**
https://github.com/kubernetes/kubernetes/pull/36707 made it necessary for the `CERT_DIR` to be accessible from within a container.  On systems with selinux enforcing, this is prohibited.

This PR updates `local-up-cluster.sh` to check for selinux enforcing mode and sets the required context on the `CERT_DIR`

@ncdc @sttts @derekwaynecarr @deads2k 

Fixes #38165

### CODE REVIEW

**Comment by ncdc** [comment]
cc @pmorie since this involves selinux/chcon

**Comment by ncdc** [comment]
cc @stevekuznetsov for bash

File: hack/local-up-cluster.sh
Comment (stevekuznetsov) [inline hack/local-up-cluster.sh]: `&>/dev/null` smells bad -- if this breaks and someone is debugging, won't they want to see the error message?

If you only conditionally want to show the message,
```sh
if ! output="$( ${CONTROLPLANE_SUDO} chcon -R -t container_file_t "${CERT_DIR}" 2>&1 )"; then
    echo "something went wrong!"
    echo "${output}"
fi
```

File: hack/local-up-cluster.sh
Comment (stevekuznetsov) [inline hack/local-up-cluster.sh]:
```sh
if ! ${CONTROLPLANE_SUDO} chcon -R -t container_file_t "${CERT_DIR}"; then
```

File: hack/local-up-cluster.sh
Comment (stevekuznetsov) [inline hack/local-up-cluster.sh]: Same comments for this as above

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: hack/local-up-cluster.sh
Comment (sttts) [inline hack/local-up-cluster.sh]: "Permissive" mode might make sense as well to avoid warnings.

**Comment by sjenning** [comment]
@stevekuznetsov @sttts thanks for the review! updates pushed.

File: hack/local-up-cluster.sh
Comment (stevekuznetsov) [inline hack/local-up-cluster.sh]: We run the risk here of aliasing `container_file_t` not existing with actual failures to `chcon` the directory. If we have / can install the SELinux tools console and use `seinfo -t` to list types available, we could make this check better. @sttts or @pmorie could comment on if that is necessary.

**Comment by sjenning** [comment]
flake #37241 
@k8s-bot gce etcd3 e2e test this issue 
@k8s-bot cvm gce e2e test this
@k8s-bot gci gce e2e test this

File: hack/local-up-cluster.sh
Comment (sjenning) [inline hack/local-up-cluster.sh]: @stevekuznetsov I see what you mean.  I think the window for that is small just because 1) we are running as root so shouldn't hit permissions errors 2) we know the directory exists and 3) we know we are on a system that likely has an selinux policy installed with the presence of getenforce.

This script is run in a variety of environments/distros so using the distro package manager in it is probably a not allowed.

IMHO I think it is good enough the way it is.  Thanks for the great bash feedback! Made the code much more pleasant to look at :)

Could I get an lgtm if you agree? All tests are green. /cc @sttts

**Review by stevekuznetsov — APPROVED** [review]
Reasoning behind not querying SELinux types directly seems OK to me.

**Comment by sttts** [comment]
Should be superseeded by https://github.com/kubernetes/kubernetes/pull/38389. Please try https://github.com/kubernetes/kubernetes/pull/38389 with selinux.

**Comment by sjenning** [comment]
#38389 makes this fix unnecessary

**Pull request closed without merging**
